tools/downloader/common.py

```diff
@@ -104,13 +105,31 @@ def check_interrupted(self):
     def interrupt(self):
         self._interrupted = True
 
+    @staticmethod
+    def _signal_message(signal_num):
+        # once Python 3.8 is the minimum supported version,
+        # signal.strsignal can be used here
+
+        signals = type(signal.SIGINT)
+
+        try:
+            signal_str = f'{signals(signal_num).name} ({signal_num})'
+        except ValueError:
+            signal_str = f'{signal_num}'
+
+        return f'Terminated by signal {signal_str}'
 
 class DirectOutputContext(JobContext):
     def print(self, value, *, end='\n', file=sys.stdout, flush=False):
         print(value, end=end, file=file, flush=flush)
 
     def subprocess(self, args, **kwargs):
-        return subprocess.run(args, **kwargs).returncode == 0
+        return_code = subprocess.run(args, **kwargs).returncode
+
+        if return_code < 0:
+            print(self._signal_message(-return_code), file=sys.stderr)
```

Comment:
Why did you use `-return_code` here?

Reply:
Have you tried it on Windows ? There might be differences in return code notations.

Reply:
@eizamaliev From https://docs.python.org/3/library/subprocess.html:

> A negative value -N indicates that the child was terminated by signal N (POSIX only).

@Fedor-Z On Windows, signals don't exist, and the return code is always non-negative.